onsp_co2/models/account_move_line.py

```diff
@@ -188,3 +203,14 @@ def can_use_product_id_carbon_value(self) -> bool:
     def get_product_id_carbon_compute_values(self) -> dict:
         self.ensure_one()
         return {"quantity": self.quantity, "from_uom_id": self.product_uom_id}
+
+    def action_recompute_analytic_line(self):
+        """
+        This method is used in the account move server action in order to recompute Co2.
+        Here we delete the analytic line and then recreate them.
+        I didn't find something that already does that.
+        """
+        for line in self:
+            line.analytic_line_ids.unlink()
+
+            line._create_analytic_lines()
```

Comment:
Is it possible to do it without the loop by any chance?

```suggestion
        self.analytic_line_ids.unlink()
        self._create_analytic_lines()
```

Reply:
We could